crates/sui-json-rpc/src/read_api.rs

```diff
@@ -292,7 +295,7 @@ impl ReadApiServer for ReadApi {
             .state
             .get_transaction_checkpoint_sequence(&digest)
             .map_err(|e| {
-                error!("Failed to get_transaction for transaction {digest:?} with error: {e:?}");
+                error!("Failed to retrieve checkpoint sequence for transaction {digest:?} with error: {e:?}");
```

Comment:
I think we can keep the change in 277-280, but on 296, we can add `unwrap_or_else` to return error that the transaction cannot be found. The idea is that no checkpoint sequence is equivalent to the transaction digest does not exist on this fullnode yet.

 I mistakenly assumed that `get_transaction_checkpoint_sequence` returns an error if the checkpoint sequence does not exist, but it actually just returns `None`.

Reply:
Ah gotcha. This was the alternative impl. but I assumed I was missing some context on why checkpoint_sequence might be some. Given the context, I can add this change